src/vs/workbench/parts/quickopen/browser/gotoLineHandler.ts

```diff
@@ -213,7 +215,31 @@ export class GotoLineHandler extends QuickOpenHandler {
 	}
 
 	getAriaLabel(): string {
-		return nls.localize('gotoLineHandlerAriaLabel', "Type a line number to navigate to.");
+
+		let label = '';
+
+		// Check we can run - (is text editor)
+		const canRun = this.canRun();
+		if (types.isUndefinedOrNull(canRun) || (typeof canRun === 'boolean' && !canRun) || typeof canRun === 'string') {
```

Comment:
Why do you check for `isUndefinedOrNull`, you own the `canRun` method and it either returns `true` or `string`. It feels like this if can be simplified.

Reply:
I "borrowed"  the logic from QuickOpenController, it seems to be conservative covering the three cases so I followed the same:

```ts
private handleSpecificHandler(handlerDescriptor: QuickOpenHandlerDescriptor, value: string, token: CancellationToken): Promise<void> {
		return this.resolveHandler(handlerDescriptor).then((resolvedHandler: QuickOpenHandler) => {

			// Remove handler prefix from search value
			value = value.substr(handlerDescriptor.prefix.length);

			// Return early if the handler can not run in the current environment and inform the user
			const canRun = resolvedHandler.canRun();
			if (types.isUndefinedOrNull(canRun) || (typeof canRun === 'boolean' && !canRun) || typeof canRun === 'string') {
				const placeHolderLabel = (typeof canRun === 'string') ? canRun : nls.localize('canNotRunPlaceholder', "This quick open handler can not be used in the current context");
```

I did tweak for this and your related, below.